### ISSUE TITLE
Erroneous results returned after failed calculation

### ISSUE THREAD

**Issue by ianhbell** [usnistgov/REFPROP-issues#447]
When an input causes a failure (here a fatal one), subsequent calls to ``ALLPROPSdll`` don't retain the error code and message and an erroneous value is merrily returned without any warning.

```python
"""Script to demonstrate issue with ``ALLPROPSdll`` returning values without errors when erroneous inputs are used.

Issue reported on:
    - SYSTEM: linux
    - REFPROP: 10.0.0.95d
"""

import os
import sys
from ctREFPROP.ctREFPROP import REFPROPFunctionLibrary


RPPREFIX = os.getenv("RPPREFIX")
RP = REFPROPFunctionLibrary(RPPREFIX)
RP.SETPATHdll(RPPREFIX)

print("SYSTEM", sys.platform)
print("REFPROP", RP.REFPROPdll("", "", "DLL#", 0, 0, 0, 0, 0, [1.0]).hUnits)
print("")

# inputs
mix_name = "AIR"
temperature = 99999
i_units = RP.GETENUMdll(1, "SI").iEnum

# set fluid
mixture_set = RP.SETMIXTUREdll(mix_name)
if mixture_set.ierr >= 100:
    raise ValueError("Error setting mixture")

# compute saturation data; other properties are affected, e.g. "FC", but demonstrating chemical potentials (CPOT) here
r = RP.REFPROPdll("", "TQ", "D", i_units, 0, 0, temperature, 0, mixture_set.z)
print("I_ERR", r.ierr)
print("H_ERR", r.herr)

density = r.Output[0]
result = RP.ALLPROPSdll("CPOT", i_units, 0, 0, temperature, density, mixture_set.z)

# BUG: Returns output values and no error message despite erroneous input.
print("OUTPUT", result.Output[0:3])
print("I_ERR", result.ierr)
print("H_ERR", result.herr)

```
yields
```
SYSTEM linux
REFPROP 10.0.1.3

I_ERR 121
H_ERR [TQFLSH error 121] Temperature input is greater than the critical or maxcondentherm temperature: T = 99999.0 K, Tc = 199.224 K.
OUTPUT array('d', [-345319.17197207815, -345319.17197207815, -345319.17197207815])
I_ERR -852
H_ERR [ALLPROPS warning -852] Additional properties are included in the output array, the input string and output array do not line up.
```

**Comment by ianhbell** [comment]
Actually the problem is more general than I thought initially. Arguments to ``ALLPROPSdll`` that are extremely far out of range (temperature of 99999 and density of -9999) don't yield a proper error code. Same issue in development version 10.0.1.8

```python
import os
import sys
from ctREFPROP.ctREFPROP import REFPROPFunctionLibrary

RPPREFIX = os.getenv("RPPREFIX")
RP = REFPROPFunctionLibrary(RPPREFIX)
RP.SETPATHdll(RPPREFIX)

print("SYSTEM", sys.platform)
print("REFPROP", RP.REFPROPdll("", "", "DLL#", 0, 0, 0, 0, 0, [1.0]).hUnits)
print("")

# inputs
mix_name = "AIR"
temperature = 99999
i_units = RP.GETENUMdll(1, "SI").iEnum

# set fluid
mixture_set = RP.SETMIXTUREdll(mix_name)
if mixture_set.ierr >= 100:
    raise ValueError("Error setting mixture")

density = -9999
result = RP.ALLPROPSdll("CPOT", i_units, 0, 0, temperature, density, mixture_set.z)

# BUG: Returns output values and no error message despite erroneous input.
print("OUTPUT", result.Output[0:3])
print("I_ERR", result.ierr)
print("H_ERR", result.herr)
```
yields
```
SYSTEM win32
REFPROP 10.0.0.02

OUTPUT array('d', [-345319.17197207815, -345319.17197207815, -345319.17197207815])
I_ERR -852
H_ERR [ALLPROPS warning -852] Additional properties are included in the output array, the input string and output array do not line up.
```

**Comment by EricLemmon** [comment]
Allprops was not calling the routine to check the limits on the temperature and density sent to the routine.  That has been fixed now and pushed up.

**Comment by ianhbell** [comment]
Confirmed. Fixed in 10.0.1.14
